File: abelkhan_type_csharp_component_msgpack/component/hub/centerproxy.cs
/*
 * centerproxy
 * qianqians
 * 2020/6/4
 */

namespace abelkhan
{
    public class centerproxy
    {
        public bool is_reg_center_sucess;
        private center_caller _caller;
        private hub_call_center_caller _hub_call_center_caller;

        public centerproxy(Ichannel ch, modulemng modules)
        {
            is_reg_center_sucess = false;
            _caller = new center_caller(ch, modules);
            _hub_call_center_caller = new hub_call_center_caller(ch, modules);
        }

        public void reg_hub(string name, string hub_type, string ip, ushort port)
        {
            _caller.reg_server("hub", hub_type, name, ip, port);
        }

        public void closed()
        {
            _hub_call_center_caller.closed();
        }
    }
}

File: abelkhan_type_component/component/hub/centerproxy.ts
import abelkhan = require('../../abelkhan_type/ts/abelkhan');
import center_protcol = require('../../protcol/ts/center');

export class centerproxy{
    public is_reg_center_sucess : boolean;
    private center_caller : center_protcol.center_caller;
    private hub_call_center_caller : center_protcol.hub_call_center_caller;
    constructor(ch:abelkhan.Ichannel, modules:abelkhan.modulemng){
        this.is_reg_center_sucess = false;
        this.center_caller = new center_protcol.center_caller(ch, modules);
        this.hub_call_center_caller = new center_protcol.hub_call_center_caller(ch, modules);
    }

    public reg_hub(name:string, ip:string, port:number){
        this.center_caller.reg_server("hub", name, ip, port);
    }

    public closed(){
        this.hub_call_center_caller.closed();
    }
}